File: Python/플로이드.py
import sys
input = sys.stdin.readline
INF = 100001


def floydWarshall(n):
    dp = [[INF*100]*(n+1) for _ in range(n+1)]

    # 초기화
    for i in range(1, n+1):
        for j in range(1, n+1):
            if graph[i][j] != INF:
                dp[i][j] = graph[i][j]

    # 플로이드
    for k in range(1, n+1):
        for i in range(1, n+1):
            for j in range(1, n+1):
                if dp[i][k] + dp[k][j] < dp[i][j]:
                    dp[i][j] = dp[i][k] + dp[k][j]

    # 출력
    for i in range(1, n+1):
        for j in range(1, n+1):
            if i == j or dp[i][j] == INF*100:
                print(0, end=" ")
            else:
                print(dp[i][j], end=" ")
        print()


if __name__ == '__main__':
    n = int(input())
    m = int(input())

    graph = [[INF]*(n+1) for _ in range(n+1)]

    for _ in range(m):
        a, b, c = map(int, input().split())
        graph[a][b] = min(graph[a][b], c)

    floydWarshall(n)

File: C++/플로이드.cpp
#include <iostream>
using namespace std;
#define INF 10000001

int graph[101][101];

int main()
{
	ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);

	int n, m;
	cin >> n >> m;

	int i, j, k;
	for (i = 0; i < 101; i++) {
		for (j = 0; j < 101; j++) graph[i][j] = INF;
	}

	for (i = 0; i < m; i++) {
		int a, b, c;
		cin >> a >> b >> c;
		graph[a][b] = min(graph[a][b], c);
	}

	// Solution
	for (k = 1; k < n+1; k++) {
		for (i = 1; i < n + 1; i++) {
			for (j = 1; j < n + 1; j++) {
				graph[i][j] = min(graph[i][j], graph[i][k] + graph[k][j]);
			}
		}
	}

	// Print
	for (i = 1; i < n+1; i++) {
		for (j = 1; j < n + 1; j++) {
			if (i == j || graph[i][j] == INF) cout << "0 ";
			else cout << graph[i][j] << " ";
		 }
		cout << "\n";
	}

	return 0;
}
